client/ratings_reviews/components/ReviewFormModal.jsx

```diff
@@ -120,19 +138,26 @@ class ReviewFormModal extends React.Component {
   }
 
   validateForm() {
-    let valid = true;
     let properties = ['rating', 'summary', 'body', 'name', 'email', 'characteristics'];
-    properties.forEach(property => {
-      let invalidCharacteristics = (property === 'characteristics' && Object.keys(this.state[property]).length !== Object.keys(this.props.meta.characteristics).length);
-      let invalidEmail = (property === 'email' && this.state[property].indexOf('@') === -1);
-      if (invalidEmail || invalidCharacteristics || this.state[property] === '') {
-        $('#review-form-' + property).text('*required');
-        valid = false;
+    let warningText = '';
+    for (let i = 0; i < properties.length; i++) {
+      let property = properties[i];
+      if ((property === 'rating' && this.state.rating === '') ||
+      (property === 'summary' && (this.state.summary.replace(/\s+/g, 'z').length > 60 || this.state.summary === '')) ||
```

Comment:
I wonder if this could be encapsulated in a helper function to make this section more readable? Not sure if it makes sense for this or not.

Reply:
I could definitely turn this into a helper! I just need to pass several state properties as arguments.